connections/README.md

```diff
@@ -21,122 +22,164 @@ and spec status.
+        - [A Note about Interoperability](#a-note-about-interoperability)
+        - [Protocol IDs (Extended)](#protocol-ids-extended)
+    
 
 ## Overview
 
-This document describes the process of establishing connections to new peers in
-libp2p and, if necessary, adding security and stream multiplexing capabilities
-to "raw" connections provided by transport protocols.
+This document describes the process and anatomy of establishing connections to new peers in
+libp2p, alongside describing several fundamental aiding protocols and processes. 
 
-We also discuss opening new streams over an existing connection, and the
-protocol negotiation process that occurs to route traffic to the correct
-protocol handler.
+We also discuss the procol negociation processes that help set up sessions, and opening new additional connections over an existing connection.
 
-This document does not cover the establishment of "transport level" connections,
+This document does not cover the establishment of "transport level" connections, 
 for example opening "raw" TCP sockets, as those semantics are specific to each
 transport.
 
 What is covered here is the process that occurs after making the initial
-transport level connection, up to the point where "application level" streams
-are opened, and their protocols are identified and data is routed appropriately
+transport level connection, up to the point where "application level" connections
+are opened, and their protocols are identified, and data is routed appropriately
 to handler functions.
 
-
 ## Definitions
 
-A **connection** is a reliable, bidirectional communication channel between two
-libp2p peers that provides **security** and the ability to open multiple
-logically independent **streams**.
+### Protocol
+
+Protocols lay at the heart of libp2p, providing modularity by availability on both parties.
+
+Protocols are abritrary processes on both ends of the libp2p [session] that write/read to the underlying [connection].
+
+Protocols can be pretty much anything, from HTTP to Identity exchange, from [multiplexing] to security.
+
+Protocols are identified by their **Protocol ID**. This can be any string, but convention is that protocols identify themselves by forward slashes ( `/` )
+
+For example, `/mplex/1.0.0` is the Protocol ID for the [ `mplex` stream-multiplexing protocol][mplex].
+
+*A recommended markup of Protocol IDs is: `/<corresponding propery>/<protocol name>/<semantic version>` , with `corresponding property` often dropped with core libp2p-backed protocols.*
+
+An unique trait of protocols is that they're programmatically able to produce further agnostic [connections][connection] or [transports][transport], making it possible to stack features of protocols on top of eachother (see [Securing Connections](#securing-connections) and [Multiplexing Streams](#multiplexing-streams)).
+
+### Connection
+
+A connection is an abstract bi-directional pipe that carries a [protocol].
```

Comment:
This is not correct. A connection does not carry a protocol. A connection is a transport-dependent pipe that connects two peers. A connection can hold streams, each of which does associate with a protocol.

Reply:
See my above comment, this is an abstraction. Btw, a connection *does* associate itself with a protocol technically, with the default stack of `security -> mplex`, a connection technically carries a TLS protocol, which produces another connection for `mplex` to use to produce multiple connections (streams) from.